src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -53,6 +53,7 @@ QgsTemporalControllerWidget::QgsTemporalControllerWidget( QWidget *parent )
   setWidgetStateFromNavigationMode( mNavigationObject->navigationMode() );
   connect( mNavigationObject, &QgsTemporalNavigationObject::navigationModeChanged, this, &QgsTemporalControllerWidget::setWidgetStateFromNavigationMode );
   connect( mNavigationObject, &QgsTemporalNavigationObject::temporalExtentsChanged, this, &QgsTemporalControllerWidget::setDates );
+  connect( mNavigationObject, &QgsTemporalNavigationObject::temporalFrameDurationChanged, this, &QgsTemporalControllerWidget::setTimeStep );
```

Comment:
I'm a little worried here about a signal bouncing back and forward from the temporal navigation object to the widget, and then the logic in QgsTemporalControllerWidget::setTimeStep causing a change to mTimeStepsComboBox or mStepSpinBox and then that being sent back to the QgsTemporalNavigationObject. Can you put a blocker in here by setting a local "bool mBlockFrameDurationChangedSignal" in QgsTemporalControllerWidget which is set to true at the start of setTimeStep and set back to false at the end of setTimeStep, and which prevents the extra call back to setTimeStep here?

Reply:
@nyalldawson thanks for looking. Yes I will. This is always one of those gui/events-programming things that I'm not sure how to do this in a proper/save/clean way... I think the safest way would be a gui in which you set a TimeStep/QgsInterval in one go, that is there is no 'intermediate product' going to the controller when the user is busy with one of the 2 widgets involved in this... For now I will add the blocker.

Reply:
@nyalldawson trying about this, and thinking about it, I think that the 'check' in 

https://github.com/qgis/QGIS/pull/39498/files#diff-b4a4c7eb1b9251a97a6e0bdaad9d253fcedd249631167e366cd3f4c27f5789f6R181-R184

actually is enough to prevent bouncing of events?

Looking at this problem as a true Model View way: if  we would have a couple of views/widgets showing the state of the model (QgsTemporalNavigationObject) we want those to change immediately if the user changes something in another view isn't it?

I'm also trying/willing to implement your idea, but just don't fully get it...
What could/should I block from happening? The fact that a user can (via the widgets) be in the middle of setting a (non-valid intermediat) value, I cannot block is it?

Reply:
I'm not confident that that check will be sufficient, because the logic in the widget is quite complex (and I can't tell immediately how it's going to mess with the value coming from the navigation object)!